pyteal/ast/tmpl.py

```diff
@@ -8,16 +11,26 @@
     from pyteal.compiler import CompileOptions
 
 
+class TmplKind(Enum):
+    Int = 0
+    Bytes = 1
+    Addr = 2
+
+
 class Tmpl(LeafExpr):
     """Template expression for creating placeholder values."""
 
-    def __init__(self, op: Op, type: TealType, name: str) -> None:
+    _session_templates: dict[str, TmplKind] = {}
+
+    def __init__(self, op: Op, type: TealType, name: str, kind: TmplKind) -> None:
```

Comment:
Changing the API of `__init__()` should be safe as it shouldn't be called directly.